runtime/lib/ttnn/program.cpp

```diff
@@ -295,6 +295,57 @@ run(::tt::target::ttnn::ReductionOp const *op, ::ttnn::Device &device,
   }
 }
 
+template <int32_t Rank>
+static std::array<int32_t, Rank>
+vectorToArray(const std::vector<int32_t> &vec) {
+  if (vec.size() != Rank) {
+    throw std::invalid_argument("Vector size does not match array size");
+  }
+  std::array<int32_t, Rank> arr;
+  std::copy(vec.begin(), vec.end(), arr.begin());
+  return arr;
+}
+
+template <int32_t Rank>
+static ::ttnn::Tensor invoke_reshape(const ::ttnn::Tensor &tensor,
+                                     const std::vector<int32_t> &shape) {
+  return ::ttnn::reshape(tensor, vectorToArray<Rank>(shape));
```

Comment:
I believe reshape expects shape as array of unsigned ints.

Reply:
They allow -1 to be passed for one dim, so signed are required.

Reply:
You're right, I saw some definition of `reshape` where they allow only uints, but they have a wrapper around it that accepts signed ints.